mobly/controllers/android_device_lib/snippet_client_v2.py

```diff
@@ -616,6 +618,14 @@ def _stop_server(self):
           self._device,
           f'Failed to stop existing apk. Unexpected output: {out}.')
 
+  def _destroy_event_client(self):
+    """Releases all the resources acquired in `_create_event_client`."""
+    if self._event_client:
+      self._event_client.host_port = None
+      self._event_client.device_port = None
+      self._event_client.close_connection()
```

Comment:
does the order of these statements matter?
Why not close the connect before setting everything to None?

Reply:
Yes, it matters. Added comments here for the reason.